addons/mass_mailing/models/mailing_contact.py

```diff
@@ -154,6 +154,14 @@ def action_add_to_mailing_list(self):
 
         return action
 
+    def action_import(self):
+        action = self.env["ir.actions.actions"]._for_xml_id("mass_mailing.mailing_contact_import_action")
+        context = dict(self.env.context)
```

Comment:
Shouldn't we literal_eval instead ?

Reply:
Why? self.env.context it is a dict that cannot be changed. Is there a better way to add elements?

Reply:
Actually do we need this context propagation at all ?

Reply:
```py
    def action_import(self):
        action = self.env["ir.actions.actions"]._for_xml_id("mass_mailing.mailing_contact_import_action")
        if ("default_mailing_list_ids" not in action['context'] and action['context'].get("active_model") == 'mailing.list' and action['context'].get("active_ids")):
            action['context'].update({
                'default_mailing_list_ids': action["active_ids"],
            })
        return action
```

Something like that ?

Reply:
Ok we don t use it anymore. We will never have context.get("active_model") == 'mailing.list'. So I remove action_import. Ok for you ?

Reply:
I tested the flow. Now I call directly the "mass_mailing.mailing_contact_import_action" on a button in the arch. The python part was there to add an info in the context if we used a certain model. Obviously, this action is no longer used with this model. We only use this button on the "mailing.contact" template. So we don t need this custo anymore.

Reply:
works fine indeed